src/dfcx_scrapi/builders/intent.py

```diff
@@ -14,19 +14,467 @@
+            If the length is less than length of phrase list it propagate
+              the rest of the annotations automatically with no annotation.
+          repeat_count (int):
+            Indicates how many times this example was added to the intent.
+
+        Example 1: phrase = "I want to check my balance"
+        Example 2:
+          phrase = [
+              'one way', ' ticket leaving ', 'January 1',
+              ' to ', 'LAX', ' from ', 'CDG'
+          ]
+          annotations = [
+              'flight_type', '', 'departure_date',
+              '', 'arrival_city', '', 'departure_city'
+          ]
+        Example 3:
+          phrase = ["I'd like to buy a ", 'one way', ' ticket']
+          annotations = ['', 'flight_type']
+
+        Returns:
+          An Intent object stored in proto_obj
+        """
+
+        self._check_intent_exist()
+
+        # Add simple training phrase
+        if isinstance(phrase, str):
+            # Create the training phrase obj and add it to the others
+            tp = Intent.TrainingPhrase(
+                parts=[Intent.TrainingPhrase.Part(text=phrase)],
+                repeat_count=repeat_count
+            )
+            self.proto_obj.training_phrases.append(tp)
+
+            return self.proto_obj
+
+        # Add annotated training phrase
+        # Type / Error checking
+        if not (isinstance(phrase, list) and isinstance(annotations, list)):
+            raise ValueError(
+                "Both phrase and annotations should be lists."
+            )
+        if not (
+            all((isinstance(p, str) for p in phrase)) and
+            all((isinstance(a, str) for a in annotations))
+        ):
+            raise ValueError(
+                "Only strings allowed in phrase or annotations list."
+            )
+        if len(annotations) > len(phrase):
+            raise IndexError(
+                "Length of annotations list is more than phrase list!"
+            )
+
+        # Propagate the annotations list if needed
+        if len(annotations) < len(phrase):
+            annotations.extend([""] * (len(phrase) - len(annotations)))
+        # Creating parts for the training phrase
+        parts_list = []
+        for text, parameter_id in zip(phrase, annotations):
+            part = Intent.TrainingPhrase.Part(
+                text=text, parameter_id=parameter_id
+            )
+            parts_list.append(part)
+
+        # Create the training phrase obj and add it to the others
+        tp = Intent.TrainingPhrase(
+            parts=parts_list, repeat_count=repeat_count
+        )
+        self.proto_obj.training_phrases.append(tp)
+
+        return self.proto_obj
+
+
+    def add_parameter(
+        self,
+        parameter_id: str,
+        entity_type: str,
+        is_list: bool = False,
+        redact: bool = False
+    ) -> Intent:
+        """Add a parameter to Parameter attribute of proto_obj.
+
+        Args:
+          id (str):
```

Comment:
@MRyderOC your method uses `parameter_id` but your docstring says `id`.   
I think `paramater_id` is the correct way to go, so just update the docstring here.